From b58a09eabc844093a4987a28cf16d79dc2b39690 Mon Sep 17 00:00:00 2001
From: anatar818 <151121941+anatar818@users.noreply.github.com>
Date: Fri, 17 Jul 2026 12:34:28 +0530
Subject: [PATCH] RDKBACCL-1938 : DCMSetting.conf is empty due to Configurable
 WAN Interface

Reason for change: After enbaling Configurable WAN , DCM feature is not working as expected
Test Procedure: Build and flash the image ,validate DAC functionality with customised wan name
Risks: None

RDKBDEV-3470,RDKBACCL-1945 : Device Not coming Online for Configurable WAN Ethernet Based Code flow

Reason for change : In Ethernet based code flow , the Device.IP.Interface DM's are not updating with the configurred wan name when Configurable wan is enabled
Test Procedure : Customise the wan name and check the DM's , it is affecting the components like Tr069
Risks : Low
Priority : P1

Configurable wan depends on syscfg only
---
 .../TR-181/custom_sbapi/cosa_deviceinfo_apis_custom.c  |  2 +-
 source-arm/TR-181/board_sbapi/cosa_ip_apis.c           | 10 ++++++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/custom/comcast/source/TR-181/custom_sbapi/cosa_deviceinfo_apis_custom.c b/custom/comcast/source/TR-181/custom_sbapi/cosa_deviceinfo_apis_custom.c
index 60a26cb..eb6ef3d 100644
--- a/custom/comcast/source/TR-181/custom_sbapi/cosa_deviceinfo_apis_custom.c
+++ b/custom/comcast/source/TR-181/custom_sbapi/cosa_deviceinfo_apis_custom.c
@@ -214,7 +214,7 @@ CosaDmlDiGetRouterMacAddress
     )
 {
     UNREFERENCED_PARAMETER(hContext);
-#ifdef FEATURE_RDKB_XDSL_PPP_MANAGER
+#if defined(FEATURE_RDKB_XDSL_PPP_MANAGER) || defined(FEATURE_RDKB_CONFIGURABLE_WAN_INTERFACE)
     char wanPhyName[32] = {0};
     char out_value[32] = {0};
 
diff --git a/source-arm/TR-181/board_sbapi/cosa_ip_apis.c b/source-arm/TR-181/board_sbapi/cosa_ip_apis.c
index 55c8033..510ab72 100644
--- a/source-arm/TR-181/board_sbapi/cosa_ip_apis.c
+++ b/source-arm/TR-181/board_sbapi/cosa_ip_apis.c
@@ -440,6 +440,16 @@ CosaDmlIpInit
 	   g_usg_if_cfg[i].LinkType = ( TRUE == IsThisCurrentPartnerID("sky-") ) ? COSA_DML_LINK_TYPE_VlanLink : COSA_DML_LINK_TYPE_EthLink;   
 	}
 #endif
+ if (i == 0)
+ {
+        rc = strcpy_s((char *)g_ipif_names[i],sizeof(g_ipif_names[i]), G_USG_IF_NAME(i));
+#if defined(FEATURE_RDKB_CONFIGURABLE_WAN_INTERFACE) 
+        char out_value[64] = {0};
+		if (!syscfg_get(NULL, "wan_physical_ifname", out_value, sizeof(out_value)))
+			rc = strcpy_s((char *)g_ipif_names[i], sizeof(g_ipif_names[i]), out_value);         
+#endif
+ }
+ else
         rc = strcpy_s((char *)g_ipif_names[i],sizeof(g_ipif_names[i]), G_USG_IF_NAME(i));
         ERR_CHK(rc);
         if ((gDmsbIpIfLoopbackInstNum == 0) && (strcmp(G_USG_IF_NAME(i), "lo") == 0))
-- 
2.43.0

